network_plugin/__init__.py

```diff
@@ -73,12 +75,15 @@ def get_vm_ip(vca_client, ctx):
             raise cfy_exc.NonRecoverableError("Could not find vApp {0}".format(vappName))
 
         vm_info = vapp.get_vms_network_info()
-        # assume that we have 1 vm per vApp with minium 1 connection
-        connection = vm_info[0][0]
-        if connection['is_connected']:
-            return connection['ip']
-        else:
-            raise cfy_exc.NonRecoverableError("Network not connected")
+        # assume that we have 1 vm per vApp
+        for connection in vm_info[0]:
+            if connection['is_connected'] and connection['is_primary']:
```

Comment:
First we need to merge changes to pyvcloud. Right now get_vms_network_info() doesn't return info on primary interface.